hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java

```diff
@@ -650,11 +650,12 @@ public long getUsedNamespace() {
    * keys if key prefix is null.
    *
    * @param keyPrefix Bucket prefix to match
+   * @param delimiter A delimiter to group matched keys
    * @return {@code Iterator<OzoneKey>}
    */
-  public Iterator<? extends OzoneKey> listKeys(String keyPrefix)
-      throws IOException {
-    return listKeys(keyPrefix, null);
+  public Iterator<? extends OzoneKey> listKeys(String keyPrefix,
+      String delimiter) throws IOException {
+    return listKeys(keyPrefix, delimiter, null);
```

Comment:
Hey @k5342 ! Following your code changes, I have a small question: would the method listKeys() change the result for OBS type of bucket? Seemed like it'll not, would you mind explaining for me a bit? 
That'd be very helpful for me understanding this change.  Thank you!

Reply:
Hi @DaveTeng0, this change will not change the result for OBS buckets (the delimiter param is ignored for now).

Reply:
I also modified both KeyIterator and KeyIteratorFSO because the `delimiter` field is in the S3 specification, so I thought it's preferred to put the param in the base class KeyIterator.